packages/nuka/src/carousel.tsx

```diff
@@ -113,116 +114,117 @@ export const Carousel = (rawProps: CarouselProps): React.ReactElement => {
 
   const carouselRef = innerRef || carouselEl;
 
-  const getNextIndex = useCallback(
-    (to?: number) => {
-      const index = to ?? currentSlide;
-      if (index < 0) {
-        return index + count;
-      }
-      if (index === count) {
-        return 0;
-      }
-      return index;
-    },
-    [count, currentSlide]
-  );
+  const goToSlide = useCallback(
+    (targetSlideIndex: number) => {
+      // Boil down the target index (-Infinity < targetSlideIndex < Infinity) to
+      // a user-friendly index (0 ≤ targetSlideIndex < slideCount)
+      const userFacingIndex =
+        ((targetSlideIndex % slideCount) + slideCount) % slideCount;
+
+      const slideChanged = targetSlideIndex !== currentSlide;
+      slideChanged && beforeSlide(slide, userFacingIndex);
 
-  const moveSlide = useCallback(
-    (to?: number) => {
-      const nextIndex = getNextIndex(to);
-      typeof to === 'number' && beforeSlide(slide, nextIndex);
+      // if animation is disabled decrease the speed to 40
+      const msToEndOfAnimation = !disableAnimation ? propsSpeed || 500 : 40;
 
-      !disableAnimation && setAnimationEnabled(true);
+      if (!disableAnimation) {
+        setIsAnimating(true);
 
-      if (typeof to === 'number') {
-        setCurrentSlide(to);
+        clearTimeout(animationEndTimeout.current);
+        animationEndTimeout.current = setTimeout(() => {
+          if (!isMounted.current) return;
+          setIsAnimating(false);
+        }, msToEndOfAnimation);
```

Comment:
Interesting idea. We rely on transitions for the smooth animation in the library, so [`transitionend`](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event)/[`transitioncancel`](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitioncancel_event) would be the ones to use. I'll keep it in mind as I refactor stuff in the library, but right now it may create a few more edge cases than it eliminates, given the current setup.